main.tf

```diff
@@ -70,13 +70,30 @@ module "efs" {
   vpc_id             = "${var.vpc_id}"
   subnets            = "${var.private_subnets}"
   availability_zones = "${var.availability_zones}"
-  security_groups    = ["${module.elastic_beanstalk_environment.security_group_id}"]             # EB/EC2 instances are allowed to connect to the EFS
   zone_id            = "${var.zone_id}"
+
+  # EB/EC2 instances and DataPipeline instances are allowed to connect to the EFS
+  security_groups = ["${module.elastic_beanstalk_environment.security_group_id}", "${module.efs_backup.security_group_id}"]
+}
+
+# EFS backup to S3
+module "efs_backup" {
+  source                             = "git::https://github.com/cloudposse/terraform-aws-efs-backup.git?ref=tags/0.3.2"
+  name                               = "${var.name}"
+  stage                              = "${var.stage}"
+  namespace                          = "${var.namespace}"
+  region                             = "${var.aws_region}"
+  vpc_id                             = "${var.vpc_id}"
+  efs_mount_target_id                = "${element(module.efs.mount_target_ids, 0)}"
```

Comment:
Will this work consistently? Won't it depend on if the DataPipeline instance is launched in the same subnet as that `mount_target_id`? e.g. it may work sometimes, but other times the mount may fail.

Reply:
That's interesting.

Subnets within a VPC are configured to allow communication by default (even if in different Availability Zones?).

And we use the private IP address of the `EFS` for `DataPipeline` to connect.

Other items that might prohibit communication between subnets include network ACLs (which allow this communication by default).

Security Groups might also prohibit communication between instances, but we explicitly allow the `DataPipeline` security group to connect to the `EFS` security group.

Anything else?

Reply:
From an availability perspective, the DataPipeline *technically* use the endpoint in it's AZ (e.g. subnet). I don't know how we'd do with the information we currently have available.

Reply:
Datapipeline instance is being created in the first [subnet](https://github.com/cloudposse/terraform-aws-efs-backup/blob/e5e377babce46dd431b5d454ed1d288e4ea4d068/cloudformation.tf#L32) returned by [aws_subnet_ids](https://github.com/cloudposse/terraform-aws-efs-backup/blob/3ce962656f1497a8f17cd9cf1491ee5b930746d2/network.tf#L7) datasource.
We may need to be sure, that first `mount_target_id` returned by `element(module.efs.mount_target_ids, 0)` is in the same AZ as first `subnet` of [aws_subnet_ids](https://github.com/cloudposse/terraform-aws-efs-backup/blob/3ce962656f1497a8f17cd9cf1491ee5b930746d2/network.tf#L7) datasource.

Or we may need to update [terraform-aws-efs-backup](https://github.com/cloudposse/terraform-aws-efs-backup) to calculate `subnet_id` based on current [mount_target_id](https://www.terraform.io/docs/providers/aws/d/efs_mount_target.html).
Thus make `efs_backup` per `mount_target_id`.

Reply:
This concept needs to be renewed.
We should not even use an individual mount point DNS name (or IP).
`EFS` generates DNS name for the entire `EFS` regardless of the AZ/subnet.
This needs to be used everywhere.
AWS will resolve the DNS name to the correct IP address of the CORRECT mount point depending on from which AZ/subnet the instance is requesting access to the `EFS`.

Reply:
seems reasonable to me to make this change. @comeanother thoughts?

Reply:
Currently module serves for our particular case, when we have to use `IP address` instead of `DNS Name` and we should add additional `Security Group Rule` to existing EFS `Security Group` instead of adding additional `Security Group` to EFS.

I agree, that this case is very rare or occasional, but there is no guarantee, that we won't face it anymore.

I agree, that using `var.efs_host` and this update will simplify module and we won't need neither `efs_mount_target_id` nor `security_group.tf` at all. Because we will have proper `DNS name`.

But in this way we should remember proper module tag, where it is possible to use IP addresses, or we should keep in mind, that the module would require additional operations in order to use it with `IP addresses`.

I'm torn. I'm leaning to adding new capabilities without removing old ones, but I'm also leaning to simplification.

I'd like to accept @aknysh proposition and rework the module (implement `var.efs_host`, remove `efs_mount_target_id` and `security_group.tf` as they become unnecessary) and add comprehensive example regarding `IP address` case and `Modifying existing EFS Security Group instead of attaching new Security Group to EFS`.

Reply:
We might need move this discussion to [terraform-aws-efs-backup](https://github.com/cloudposse/terraform-aws-efs-backup) issues, or PR description.